.github/workflows/publish-pypi.yml

```diff
@@ -64,14 +77,66 @@ jobs:
         # Add PyPi index to the pip install command so that any dependencies not on TestPyPI are installed from PyPI
         # For example, anndata >= 0.9.0 is not on TestPyPI, so it is installed from PyPI
         run: |
+          python -m venv venv
+          source venv/bin/activate
           pip install --no-cache-dir -i https://test.pypi.org/simple/ --extra-index-url https://pypi.org/simple cz-benchmarks
           python -c "import czbenchmarks"
           pip freeze | grep "cz-benchmarks==${{ steps.get_version.outputs.VERSION }}"
 
+  publish-prod-pypi:
+    # This job will only run if the test-pypi job is successful
+    needs: publish-test-pypi
+    # Only run if done for an official release, not via a triggered workflow, which is used for testing
+    if: github.event_name == 'push' && startsWith(github.ref, 'refs/tags')
+    runs-on: ubuntu-latest
+    environment: 
+      name: pypi
+      url: https://pypi.org/cz-benchmarks
+    permissions:
+      id-token: write # to authenticate as Trusted Publisher to pypi.org
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Get the version
+        id: get_version
+        run: |
+          if [[ -n "${{ github.event.inputs.version }}" ]]; then
+            echo "VERSION=${{ github.event.inputs.version }}" >> $GITHUB_OUTPUT
+          else
+            echo "VERSION=${GITHUB_REF#refs/tags/v}" >> $GITHUB_OUTPUT
+          fi
+
+      - name: Display version being published
+        run: |
+          echo "Publishing version: ${{ steps.get_version.outputs.VERSION }}"
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version-file: .python-version
+
+      - name: Cache python dependencies
+        uses: actions/cache@v4
+        with:
+          path: ~/.cache/uv
+          key: ${{ runner.os }}-python-3.10-build-${{ hashFiles('uv.lock') }}
+          restore-keys: ${{ runner.os }}-python-3.10-build-
+
+      - name: Install uv
+        run: curl -LsSf https://astral.sh/uv/install.sh | sh
```

Comment:
This is probably okay but you could also use
```
- name: Install the latest version of uv
  uses: astral-sh/setup-uv@v6
  ```
from https://github.com/astral-sh/setup-uv

Reply:
good call! I'll file a new issue, since it's not critical to the fix of this PR.